source/javascripts/controllers/_ProvProfileController.js.erb

```diff
@@ -163,28 +192,149 @@ angular.module("BitriseWorkflowEditor").controller("ProvProfileController", func
 	};
 
 	viewModel.getDetails = function(provProfile) {
-		viewModel.provProfileConfigs[provProfile.databaseID].progress.start("<%= data[:strings][:code_signing][:prov_profile][:get_details_progress][:in_progress] %>");
+		viewModel.provProfileConfigs[provProfile.databaseID].getDetailsProgress.start("<%= data[:strings][:code_signing][:prov_profile][:get_details_progress][:in_progress] %>");
 		if (viewModel.uploadProvProfilesProgress.isIdle && viewModel.uploadProvProfilesProgress.statusMessage !== null) {
 			viewModel.uploadProvProfilesProgress.reset();
 		}
 
-		requestService.getDetailsOfProvProfile(provProfile.databaseID).then(function(data) {
-			provProfile.uuid = data.uuid;
-			provProfile.expiresAt = data.expiresAt;
-			provProfile.exportType = data.exportType;
-			provProfile.teamName = data.teamName;
-			provProfile.teamID = data.teamID;
-			provProfile.bundleID = data.bundleID;
-			provProfile.developerCertificateIdentities = data.developerCertificateIdentities;
-			provProfile.provisionedDevices = data.provisionedDevices;
-			provProfile.capabilityKeys = data.capabilityKeys;
-
-			viewModel.provProfileConfigs[provProfile.databaseID].progress.success();
+		return $q(function(resolve, reject) {
+			return requestService.getDetailsOfProvProfile(provProfile.databaseID).then(function(data) {
+				provProfile.uuid = data.uuid;
+				provProfile.expiresAt = data.expiresAt;
+				provProfile.exportType = data.exportType;
+				provProfile.teamName = data.teamName;
+				provProfile.teamID = data.teamID;
+				provProfile.bundleID = data.bundleID;
+				provProfile.developerCertificateIdentities = _.map(data.developerCertificateIdentities, function(anIdentity) {
+					return {
+						statusCSSclass: "in-progress",
```

Comment:
Use Progresses here as well. Because a) there are many places where this `statusCSSclass` is set manually, which is not very good, since it is not indicated which are the accepted values; b) later I see you set a flag conditionally based on this CSS class - it would be prettier if you would set it based on a Progress's state. In the template, you can use the progress as progress.cssClass()